modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MuleInputTypeResolverAdapter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.metadata;
+
+import static java.lang.String.format;
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.metadata.MetadataContext;
+import org.mule.runtime.api.metadata.MetadataResolvingException;
+import org.mule.runtime.api.metadata.resolving.InputTypeResolver;
+import org.mule.sdk.api.metadata.NullMetadataResolver;
+
+public class MuleInputTypeResolverAdapter implements InputTypeResolver {
+
+  private final org.mule.sdk.api.metadata.resolving.InputTypeResolver delegate;
+
+  MuleInputTypeResolverAdapter(org.mule.sdk.api.metadata.resolving.InputTypeResolver delegate) {
+    this.delegate = delegate;
+  }
+
+  public static InputTypeResolver from(Object resolver) {
```

Comment:
I feel like this logic if repeated (although for the other types of resolvers) in a few places. And the null resolver part is duplicated.

Maybe have some adapter factory class that resolves this. See that for example. the static method may not even create a MuleInputTypeResolverAdapter.